docs/rules/object-property-newline.md

```diff
@@ -72,20 +71,53 @@ Another benefit of this rule is specificity of diffs when a property is changed:
 +var obj = { foo: "foo", bar: "bazz", baz: "baz" };
 ```
 
-### Optional Exception
+### Optional Exceptions
+
+The rule offers three object option.
 
-The rule offers one object option, `allowMultiplePropertiesPerLine`. If you set it to `true`, object literals such as the first two above, with all property specifications on the same line, will be permitted, but one like
+#### `allowMultiplePropertiesPerLine`
+
+If you set `allowMultiplePropertiesPerLine` to `true`, object literals such as the first two above, with all property specifications on the same line, will be permitted, but one like
 
 ```js
 const newObject = {
     a: 'a.m.', b: 'p.m.',
     c: 'daylight saving time'
 };
-
 ```
 
 will be prohibited, because two properties, but not all properties, appear on the same line.
 
+#### `treatComputedPropertiesLikeJSCS`
+
+If you set `treatComputedPropertiesLikeJSCS` to `true`, an object literal such as the one below will be permitted:
+
+```js
+const newObject = {
+    a: 1, [
+        process.argv[4]
+    ]: '01'
+};
+```
+
+Otherwise, this rule will prohibit it, because ESLint treats the opening bracket of a computed property name as part of the property specification. The JSCS rule `requireObjectKeysOnNewLine` does not, so this object option makes ESLint compatible with JSCS in this respect.
+
+#### `noCommaFirst`
+
+If you set `noCommaFirst` to `true`, an object literal such as the one below will be prohibited, even though all its property specifications are on separate lines:
+
+```js
+const newFunction = multiplier => ({
+    a: 2 * multiplier
+    , b: 4 * multiplier
+    , c: 8 * multiplier
+});
+```
+
+This object option makes the rule stricter by prohibiting one of the patterns by which you could comply with the rule. Specifically, the comma between two property specifications may not appear before the second one on the same line. The JSCS rule `requireObjectKeysOnNewLine` treats commas this way, so this object option makes ESLint compatible with JSCS in this respect.
+
+You can use the `comma-style` rule instead of this option to achieve partial JSCS compatibility, but not in combination with the `treatComputedPropertiesLikeJSCS` object option. Using the `comma-style` rule for the sole purpose of JSCS compatibility would also require you to enumerate 9 exceptions, leaving only `ObjectExpression` subject to the rule.
```

Comment:
I'm not sure I understand this part. Why can't comma-style be used in combination with `treatComputedPropertiesLikeJSCS`?

Should this be valid or invalid:

```js
/* eslint comma-style: ["error", "first"], object-property-newline: ["error", { treatComputedPropertiesLikeJSCS: true }] */
({
    foo: "bar", [
        "computed"
    ]: true
    , baz: "woo"     // <--- this leading comma here: valid or invalid?
});
```